### PR TITLE
Add open_text and open_image in orca

### PR DESCRIPTION
**Pull request by hkvision**
https://github.com/analytics-zoo/orca/issues/8
https://github.com/analytics-zoo/orca/issues/9

Can Popen to execute a cat command to read hdfs text files. Alternatively, https://pypi.org/project/pydoop/ can be used as well.

### CODE REVIEW

File: pyzoo/zoo/orca/data/utils.py [hunk 1]
```diff
@@ -85,3 +85,24 @@ def extract_one_path(file_path, file_type, context):
                           for file in os.listdir(file_path)
                           if os.path.splitext(file)[1] == "." + file_type]
     return file_paths
+
+
+def open_file(path):
+    if path.startswith("hdfs"):  # hdfs://url:port/file_path
+        from subprocess import Popen, PIPE
+        return Popen(["hadoop", "fs", "-cat", path], stdout=PIPE, encoding="utf-8").stdout
+        # Can use pydoop as well.
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/utils.py:94]: can we use `pyarrow.HdfsFile.download`?

Comment (hkvision) [reply pyzoo/zoo/orca/data/utils.py]: I changed to pyarrow. Take a look? @jason-dai 
Read text file as bytes, decode as string and split by \n; return a list of strings. For HyperSeg code, just need to change import and function name, and no other changes needed.
Built-in `open` function returns a TextIOWrapper, don't know how to have exactly the same return type for hdfs and s3...

**Comment by hkvision** [comment]
@jason-dai Finished the two utils. Take a look. Would add some unit tests afterwards.

File: pyzoo/zoo/orca/data/utils.py [hunk 2]
```diff
@@ -76,3 +76,50 @@ def get_class_name(obj):
...
         return obj.__class__.__name__
     return module + '.' + obj.__class__.__name__
+
+
+def open_file(path):
+    if path.startswith("hdfs"):  # hdfs://url:port/file_path
+        import pyarrow as pa
+        fs = pa.hdfs.connect()
+        with fs.open(path, 'rb') as f:
+            return f.read().decode("utf-8").split("\n")
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/utils.py:86]: this only works for text?

File: pyzoo/zoo/orca/data/utils.py [hunk 3]
```diff
@@ -76,3 +76,50 @@ def get_class_name(obj):
...
+
+
+def open_image(path):
+    from PIL import Image
+    if path.startswith("hdfs"):  # hdfs://url:port/file_path
+        import pyarrow as pa
+        from io import BytesIO
+        fs = pa.hdfs.connect()
+        with fs.open(path, 'rb') as f:
+            return Image.open(BytesIO(f.read()))
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/utils.py:110]: does `Image.open(f)` work? `f` is supposed to be a File object? https://arrow.apache.org/docs/python/generated/pyarrow.HdfsFile.html

Comment (hkvision) [reply pyzoo/zoo/orca/data/utils.py:110]: Seems Image.open doesn't support HdfsFile...

File: pyzoo/zoo/orca/data/utils.py (hunk 2 above)
Comment (hkvision) [reply pyzoo/zoo/orca/data/utils.py:86]: Yeah. Any other file formats we need to support?

Comment (jason-dai) [reply pyzoo/zoo/orca/data/utils.py:86]: then we may change the method to `open_text`?

Comment (hkvision) [reply pyzoo/zoo/orca/data/utils.py:86]: okay!

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.210:18888/job/ZOO-PR-Validation/3405/

**Comment by hkvision** [comment]
tested hdfs and s3 manually

**Pull request merged**